frontend/routes/account/tokens/index.tsx

```diff
@@ -0,0 +1,207 @@
+// Copyright 2024 the JSR authors. All rights reserved. MIT license.
+import { Handlers, PageProps } from "$fresh/server.ts";
+import { State } from "../../../util.ts";
+import { path } from "../../../utils/api.ts";
+import { FullUser, Token } from "../../../utils/api_types.ts";
+import { AccountLayout } from "../(_components)/AccountLayout.tsx";
+import { Head } from "$fresh/runtime.ts";
+import twas from "$twas";
+import { RevokeToken } from "./(_islands)/RevokeToken.tsx";
+import { Plus } from "../../../components/icons/Plus.tsx";
+
+interface Data {
+  user: FullUser;
+  tokens: Token[];
+}
+
+export default function AccountTokensPage(
+  { data, url }: PageProps<Data, State>,
+) {
+  const personal = data.tokens.filter((token) => token.type === "personal");
+  const sessions = data.tokens.filter((token) => token.type !== "personal");
+
+  return (
+    <AccountLayout user={data.user} active="Tokens">
+      <Head>
+        <title>
+          Your tokens - JSR
+        </title>
+      </Head>
+      <div>
+        <h2 class="text-xl mb-2 font-bold">Personal access tokens</h2>
+        <p class="text-gray-600 max-w-2xl">
+          Personal access tokens can be used to authenticate with JSR from the
+          command line or from other applications.
+        </p>
+
+        {personal.length > 0
+          ? (
+            <ul class="max-w-2xl divide-slate-200 divide-y border-t border-b border-slate-200 mt-4">
+              {personal.map((token) => <PersonalTokenRow token={token} />)}
+              <li class="py-2">
+                <a
+                  href="/account/tokens/create"
+                  class="flex items-center gap-2 text-jsr-cyan-700 hover:text-jsr-cyan-600 hover:underline outline-none focus-visible:ring-2 ring-jsr-cyan-700 ring-offset-2 rounded-sm"
+                >
+                  <Plus />
+                  Create new token
+                </a>
+              </li>
+            </ul>
+          )
+          : (
+            <div class="mt-6">
+              <p class="italic text-gray-600">
+                You have no personal access tokens.
+              </p>
+              <p class="mt-2">
+                <a
+                  href="/account/tokens/create"
+                  class="flex items-center gap-2 text-jsr-cyan-700 hover:text-jsr-cyan-600 hover:underline outline-none focus-visible:ring-2 ring-jsr-cyan-700 ring-offset-2 rounded-sm"
+                >
+                  <Plus />
+                  Create new token
+                </a>
```

Comment:
maybe use the usual button rather than text? doesnt work too well in the case there is a token, so the button should be outside the list

Reply:
I tried a button first - but it was very intrusive, especially on mobile. Can we leave it as is?

Reply:
sure